modules/bootstrapped/test/src/smithy4s/BlobSpec.scala

```diff
@@ -62,4 +55,105 @@ class BlobSpec() extends FunSuite {
     assertNotEquals(blob1.hashCode, blob3.hashCode)
   }
 
+  test("Concat works as expected") {
+    val blob = Blob("foo") ++ Blob("bar")
+    assertEquals(blob.size, 6)
+    assertEquals(blob(2), 'o'.toByte)
+    assertEquals(blob(4), 'a'.toByte)
+    java.util.Arrays
+      .equals(blob.toArray, "foo".getBytes ++ "bar".getBytes())
+  }
+
+  val all = List(
+    "Queue" -> (Blob("foo") ++ Blob("bar")),
+    "Array" -> Blob("foobar"),
+    "Buffer" -> Blob(ByteBuffer.wrap("foobar".getBytes()))
+  )
+
+  for {
+    x <- all
+    y <- all
+  } {
+    test(s"${x._1} and ${y._1} : same bytes") {
+      assert(x._2.sameBytesAs(y._2))
+    }
+  }
+
+  all.foreach { case (name, data) =>
+    test(s"$name: size") {
+      assertEquals(data.size, 6)
+    }
+
+    test(s"$name: index access") {
+      assertEquals(data(2), 'o'.toByte)
+    }
+
+    test(s"$name: out of bounds access") {
+      try {
+        val _ = data(6)
+        fail("expected exception")
+      } catch {
+        case _: IndexOutOfBoundsException => ()
+      }
+    }
+
+    test(s"$name: utf8String") {
+      assertEquals(data.toUTF8String, "foobar")
+    }
+
+    test(s"$name: toArraySliceBlob") {
+      assertEquals[Blob, Blob](
+        data.toArraySliceBlob,
+        Blob.slice("foobar".getBytes(), 0, 6)
+      )
+    }
+
+    test(s"$name: copyToArray") {
+      val target = Array.fill[Byte](6)(0)
+      data.copyToArray(target, 0, 0, data.size)
+      assert(target.sameElements("foobar".getBytes()))
+    }
+
+    test(s"$name: copyToBuffer") {
+      val target = ByteBuffer.wrap(Array.fill[Byte](6)(0))
+      data.copyToBuffer(target, 0, data.size)
+      assert(target.array.sameElements("foobar".getBytes()))
+    }
+
+    test(s"$name: copyToStream") {
+      val stream = new ByteArrayOutputStream()
+      try {
```

Comment:
the other day I discovered `Using` and [`Using.resource`](https://www.scala-lang.org/api/2.13.12/scala/util/Using$.html#resource[R,A](resource:R)(body:R=%3EA)(implicitreleasable:scala.util.Using.Releasable[R]):A) to do this kind of resource management

Reply:
Thanks. I was under the impression it was a 2.13+ only thing 

[b50d8d4](https://github.com/disneystreaming/smithy4s/pull/1211/commits/b50d8d4b22cf0095e0b4e4f0e92c505ffb3d6540)